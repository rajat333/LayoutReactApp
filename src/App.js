import React, {Component} from 'react';
import {Router, Route, Switch} from 'react-router-dom';
import {history} from './_helpers';
import './App.css';
import asyncComponent from "./AsyncComponent";
import PrivateRoute from "./components/PrivateRoute/PrivateRoute";
import userService from "./services/userService";
// lazy loading import Home from "./components/Home/Home"; import Login from
// "./components/Login/Login"; import ContactUs from
// "./components/ContactUs/ContactUs"; import AboutUs from
// "./components/AboutUs/AboutUs"; import Register from
// './components/Register/Register'; import Dashboard from
// "./components/Dashboard/Dashboard"; import Profile from
// './components/Profile/Profile'; import Ticket from
// './components/Ticket/RaiseTicket';
// import LogOut from './components/LogOut/LogOut';
// import SingleUser from './components/Dashboard/SingleUser/SingleUser';
const Home = asyncComponent(
    () => import ('./components/Home/Home').then(module => module.default)
)
const Dashboard = asyncComponent(
    () => import ('./components/Dashboard/Dashboard').then(module => module.default)
)

class App extends Component {

    constructor(props) {
        super(props);
        // documentation https://github.com/ReactTraining/history
        history.listen((location, action) => {
            // clear alert on location change dispatch(alertActions.clear());
            console.log(
                "...location...action...",
                location,
                action,
                userService.isUserLogIn()
            );
        });
    }

    componentWillMount() {
        // console.log("..component will mount...");
    }

    componentWillReceiveProps(nextProps) {
        // console.log("...nextProps...",nextProps);
    }
    render() {
        return (
            <div className="App">
                <Router history={history}>
                    <div>
                        <Switch>
                            <Route exact={ true } path="/" component={Home}/>
                            <PrivateRoute path="/dashboard" component={Dashboard}/>
                            <Route component={Home}/>
                        </Switch>

                    </div>

                </Router>
            </div>
        );
    }
}

export default App;
