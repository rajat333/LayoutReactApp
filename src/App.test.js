import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <App/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
// describe('Addition', () => {
//   it('knows that 2 and 2 make 4', () => {
//     expect(2 + 2).toBe(4);
//   });
// });


test('adds 1 + 2 to equal 3', () => {
  expect(sum(1, 2)).toBe(3);
});

var sum = (a,b)=>{ return (a+b); }