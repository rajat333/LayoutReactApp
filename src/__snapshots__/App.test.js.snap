// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="not-found"
    >
      <div
        className="Header"
      >
        <nav
          className="navbar navbar-inverse container-fluid navbar-fixed-top"
        >
          <div>
            <div
              className="navbar-header"
            >
              <button
                aria-expanded="false"
                className="navbar-toggle collapsed"
                data-target="#bs-example-navbar-collapse-1"
                data-toggle="collapse"
                type="button"
              >
                <span
                  className="sr-only"
                >
                  Toggle navigation
                </span>
                <span
                  className="icon-bar"
                />
                <span
                  className="icon-bar"
                />
                <span
                  className="icon-bar"
                />
              </button>
              <a
                className="navbar-brand"
                href="/"
                onClick={[Function]}
              >
                Demo React App
              </a>
            </div>
            <div
              className="collapse navbar-collapse"
              id="bs-example-navbar-collapse-1"
            >
              <ul
                className="nav navbar-nav"
              >
                <li
                  className=""
                >
                  <a
                    href="/"
                    onClick={[Function]}
                  >
                    <span
                      className="glyphicon glyphicon-home"
                    />
                    Home
                  </a>
                </li>
                <li>
                  <a
                    href="about"
                    onClick={[Function]}
                  >
                    <span
                      className="glyphicon glyphicon-king"
                    />
                    About
                  </a>
                </li>
              </ul>
              <form
                className="navbar-form navbar-left"
                role="search"
              >
                <div
                  className="form-group-sm"
                >
                  <div
                    className="input-group"
                  >
                    <input
                      className="form-control"
                      onChange={[Function]}
                      placeholder="Search"
                      type="text"
                      value=""
                    />
                    <span
                      className="input-group-btn"
                    >
                      <button
                        className="btn btn-success btn-sm"
                        onClick={[Function]}
                      >
                        Go!
                      </button>
                    </span>
                  </div>
                </div>
              </form>
              <ul
                className="nav navbar-nav navbar-right"
              >
                <li>
                  <a
                    href="contactus"
                    onClick={[Function]}
                  >
                    <span
                      className="glyphicon glyphicon-phone"
                    />
                    Contact Us
                  </a>
                </li>
                <li>
                  <a
                    className="active"
                    href="login"
                    onClick={[Function]}
                  >
                    <span
                      className="glyphicon glyphicon-log-in"
                    />
                    Login
                  </a>
                </li>
                <li>
                  <a
                    href="register"
                    onClick={[Function]}
                  >
                    <span
                      className="glyphicon glyphicon-user"
                    />
                    Register
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
        <div
          className="frontbackground"
          style={
            Object {
              "backgroundImage": "url(/images/frontbg.jpg)",
              "height": "400px",
              "marginTop": "51px",
              "width": "100%",
            }
          }
        />
      </div>
      <h1>
        404 Not Found
      </h1>
    </div>
  </div>
</div>
`;
